lib/harvestdor/indexer.rb

```diff
@@ -63,11 +63,16 @@ def harvest_and_index(each_options = { in_threads: 4 })
     end
 
     def resources
-      druids.map do |x|
-        Harvestdor::Indexer::Resource.new(self, x)
-      end.map do |x|
-        [x, (x.items if x.collection?)]
-      end.flatten.uniq.compact
+      return to_enum(:resources) unless block_given?
+
+      druids.each do |x|
+        resource = Harvestdor::Indexer::Resource.new(self, x)
+        yield resource
+
+        resource.items.each do |member|
+          yield member
+        end if resource.collection?
+      end
```

Comment:
I guess I'm wondering if it might read something like this  (and i freely admit I'm not skilled at yields and procs and lambdas and passing blocks around):
```ruby
# return an enumerable collection of resources which includes each single item in the druids list, 
#   and all the items belonging to each collection id in druids.
def resource
  return to_enum(:resources) unless block_given?

  druids.each do |id|
    # create a resource if it's an item 
    resource = Harvestdor::Indexer::Resource.new(self, id)
    yield resource

   # fetch item ids if it's a collection, and include those items in returned enum
   resource.items.each do |member|
    yield member
  end if resource.collection?
end

```